core/lib/zksync_core/src/eth_watch/mod.rs

```diff
@@ -212,7 +212,7 @@ pub async fn start_eth_watch(
     )
     .await;
 
-    Ok(tokio::spawn(async move {
-        eth_watch.run(pool, stop_receiver).await
-    }))
+    Ok(tokio::spawn(
+        async move { eth_watch.run(stop_receiver).await },
```

Comment:
Btw you probably don't need `async move { .. }` block here.

Reply:
why

Reply:
but `move` is still needed

Reply:
Just change `run` method to take `mut self` instead of `&mut self`.